project/MimaExcludes.scala

```diff
@@ -36,6 +36,8 @@ object MimaExcludes {
 
   // Exclude rules for 3.0.x
   lazy val v30excludes = v24excludes ++ Seq(
+    // [SPARK-25959] GBTClassifier picks wrong impurity stats on loading
+    ProblemFilters.exclude[IncompatibleResultTypeProblem]("org.apache.spark.ml.classification.GBTClassificationModel.setImpurity"),
```

Comment:
a note for reviewers: setting this parameter on the model doesn't make any sense. We may also override the `setImpurity` method in order to throw an exception as done in the `GBTClassifier`...